lib/modules/navigationMiddleware.js

```diff
@@ -10,7 +10,7 @@ export default {
         const { dispatch, getState } = store;
 
         for (let route of routes) {
-          const [url, handler] = route;
+          const [url, handler, name] = route;
```

Comment:
this feels a bit odd to me. the `name` property isn't really a property of the url, so it wouldn't be used to distinguish between the routes to pick a handler. Consider:

```es6
export default [
  ['/foo/:thing', ControllerOne, 'thing1'],
  ['/foo/:thing', ControllerTwo, 'thing2'],
];
```

The names (`thing1` vs `thing2`) aren't really going to be used to differentiate between picking the two controllers. But looking at the way the routes are declared, a programmer might expect that the third parameter in a route is used by the router for something.

Could we instead make the names a property of the controller?